bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/impl/ProductGraphqlServicImpl.java

```diff
@@ -0,0 +1,151 @@
+/*
+ *   Copyright 2019 Adobe Systems Incorporated
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.adobe.cq.commerce.core.components.internal.impl;
+
+import com.adobe.cq.commerce.core.components.internal.models.v1.product.AssetImpl;
+import com.adobe.cq.commerce.core.components.models.product.Asset;
+import com.adobe.cq.commerce.core.components.service.ProductGraphqlService;
+import com.adobe.cq.commerce.graphql.client.GraphqlClient;
+import com.adobe.cq.commerce.graphql.client.GraphqlRequest;
+import com.adobe.cq.commerce.graphql.client.GraphqlResponse;
+import com.adobe.cq.commerce.magento.graphql.*;
+import com.adobe.cq.commerce.magento.graphql.gson.Error;
+import com.adobe.cq.commerce.magento.graphql.gson.QueryDeserializer;
+import org.osgi.service.component.annotations.Component;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ *
+ *GraphQlService extracted from ProductModel
+ * It could be one service to cater to all component , carrying all the service methods
+ * ToDO: Refactoring
+ *
+ */
+
+@Component(service = ProductGraphqlService.class)
+public class ProductGraphqlServicImpl implements ProductGraphqlService {
+
+    private static final String PRODUCT_IMAGE_FOLDER = "catalog/product";
+    private String mediaBaseUrl;
+
+
+    @Override
+    public List<Asset> filterAndSortAssets(List<MediaGalleryEntry> assets) {
```

Comment:
If you switch the query to use swatch image this all is not needed. It is wrong anyway for the current use case because the position only has of the truth, there might be one particular image tagged as to be used as small or swatch image in Magento.